src/fields.jl

```diff
@@ -198,12 +229,18 @@ end
 #####
 
 δxᶜ(i, f::Field1D{Face})   = f.data[i+1] - f.data[i]
+δxᶜ(i, f::Field1D{Centre}) = f.data[i+1] - f.data[i-1]
 δxᶠ(i, f::Field1D{Centre}) = f.data[i] - f.data[i-1]
+δxᶠ(i, f::Field1D{Face})   = f.data[i+1] - f.data[i-1]
```

Comment:
great addition, but needs a test!

Reply:
Tests for these are already present, since I have 4 tests for derivatives where the input and output live on same grid.

Reply:
Or I misunderstood you?

Reply:
I simply claimed that this was not tested. It's easier if you point me to the part of the code that test this.

Reply:
On line 121 of runtests.jl, 
`@test test_∂x(∂u_on_u_actual1D, ∂htest1D, u1D)` calls `∂x!(output::Field1D{Face}, input::Field1D{Face, Grid1D{Periodic}})`, which calls this this newly added line.

Reply:
well, codecov disagrees with you:
https://codecov.io/gh/ClimateFluidPhysics-ANU/MixedLayerThermoclineDynamics.jl/src/2313e6ff82ff383874a345ce4492a320b792398b/src/fields.jl
let me have a closer look.

Reply:
OK, now I see.

You added these lines in this PR:
```
δxᶜ(i, f::Field1D{Centre}) = f.data[i+1] - f.data[i-1]
δxᶠ(i, f::Field1D{Face})   = f.data[i+1] - f.data[i-1]
```
right?

They are never called though.

Perhaps think clearly what you wanna do and make sure the code reflects on whatever is in your head? Or explain here in a concrete manner what you wanna do and we can discuss how to do it?

(I can fix it, but it'll be better if you do...)